### PR TITLE
Fixes #730

### LINKED ISSUE #730: RTL Support |  Carousel Problem
### Versions

| package  | version |
| -------- | ------- |
| `react`  | `16.11` |
| `rsuite` | `4.1.1` |

### What is the expected behavior?
The **Carousel** should function as it currently does in the **LTR** scenario except that its **layout** should be **flipped** properly in the **RTL** case.

### What is the current behavior?
The **Carousel** does **not** render correctly on **RTL** pages. 

### What are the steps to reproduce?
go to documentation > components > **Carousel**.
switch to **RTL**, you will see it is not working, slides don't appear

### PR DESCRIPTION
**Pull request by MohammadYounes**
Ensures correct Carousel behavior in RTL mode.

### CODE REVIEW

File: src/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -64,90 +65,98 @@ class Carousel extends React.Component<CarouselProps, CarouselState> {
...
-      </Animation.Transition>
+          return (
+            <Animation.Transition
+              enteredClassName={addPrefix('last')}
+              exitingClassName={addPrefix('reset')}
+              in={last}
+            >
+              <Component className={classes} {...unhandled}>
+                <div className={addPrefix('content')}>
+                  <div className={addPrefix('slider')} style={sliderStyles}>
```
Comment (simonguo) [inline src/Carousel/Carousel.tsx]: The scope of the context should be as small as possible.
```jsx
<IntlContext.Consumer>
  {context => {
    const activeRatio = `${!vertical && context?.rtl ? '' : '-'}${(100 / count) * active}%`;
    const sliderStyles = {
      [lengthKey]: `${count * 100}%`,
      transform: vertical
        ? `translate3d(0, ${activeRatio} ,0)`
        : `translate3d(${activeRatio}, 0 ,0)`
    };
    return (
      <div className={addPrefix('slider')} style={sliderStyles}>
        {items}
      </div>
    );
  }}
</IntlContext.Consumer>
```

File: src/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -64,90 +65,98 @@ class Carousel extends React.Component<CarouselProps, CarouselState> {
...
+          );
 
-    const activeRatio = `-${(100 / count) * active}%`;
-    const sliderStyles = {
-      [lengthKey]: `${count * 100}%`,
-      transform: vertical
-        ? `translate3d(0, ${activeRatio} ,0)`
-        : `translate3d(${activeRatio}, 0 ,0)`
-    };
+          const activeRatio = `${!vertical && context.rtl ? '' : '-'}${(100 / count) * active}%`;
```
Comment (simonguo) [inline src/Carousel/Carousel.tsx]: Need to determine if the context is undefined.
```diff
- const activeRatio = `${!vertical && context.rtl ? '' : '-'}${(100 / count) * active}%`;
+ const activeRatio = `${!vertical && context?.rtl ? '' : '-'}${(100 / count) * active}%`;
```

File: src/Carousel/Carousel.tsx [hunk 3]
```diff
@@ -64,90 +65,98 @@ class Carousel extends React.Component<CarouselProps, CarouselState> {
...
+                  <ul>{labels}</ul>
+                </div>
+              </Component>
+            </Animation.Transition>
+          );
+        }}
+      </IntlContext.Consumer>
     );
   }
 }
```
Comment (simonguo) [inline src/Carousel/Carousel.tsx]: Need to modify the `index.tsx` file
```diff
import Carousel from './Carousel';
+import withLocale from '../IntlProvider/withLocale';

- export default Carousel;
+ export default withLocale([])(Carousel);
```

File: src/Carousel/Carousel.tsx (hunk 1 above)
Comment (MohammadYounes) [reply src/Carousel/Carousel.tsx]: @simonguo Done

**Comment by MohammadYounes** [comment]
@simonguo Thanks for the notes! All done.

**Comment by simonguo** [comment]
@MohammadYounes  
👌 This PR will be merged before the next fixed release.

**Pull request merged**